src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/UseSpanClearInsteadOfFillTests.cs

```diff
@@ -0,0 +1,250 @@
+
+namespace Microsoft.NetCore.Analyzers.Performance.UnitTests
+{
+    public class UseSpanClearInsteadOfFillTests
+    {
+        [Fact]
+        public async Task TestCodeFix()
+        {
+            string source = @"
+using System;
+
+class C
+{
+    void M(Span<byte> span)
+    {
+        [|span.Fill(0)|];
+    }
+}
+";
+            string expected = @"
+using System;
+
+class C
+{
+    void M(Span<byte> span)
+    {
+        span.Clear();
+    }
+}
+";
+            await TestCS(source, expected);
+        }
+
+        [Theory]
+        [InlineData("int", "0")]
+        [InlineData("int", "1 - 1")]
+        [InlineData("long", "0")]
+        [InlineData("double", "0")]
+        [InlineData("double", "0.0")]
+        [InlineData("nint", "0")]
+        [InlineData("int", "(int)-0.0")]
+        [InlineData("object", "null")]
+        [InlineData("string", "null")]
+        [InlineData("int?", "null")]
+        [InlineData("int", "default")]
+        [InlineData("int?", "default")]
+        [InlineData("DateTime", "new DateTime()")]
+        [InlineData("DateTime", "default")]
+        [InlineData("DateTime", "default(DateTime)")]
+        [InlineData("DayOfWeek", "DayOfWeek.Sunday")]
+        [InlineData("DayOfWeek", "(DayOfWeek)0")]
+        public async Task TestDefaultValue(string type, string value)
+        {
+            string source = $@"
+using System;
+
+class C
+{{
+    void M(Span<{type}> span)
+    {{
+        [|span.Fill({value})|];
+    }}
+}}
+";
+            await TestCS(source);
+        }
+
+        [Theory]
+        [InlineData("int", "1")]
+        [InlineData("float", "-0.0f")]
+        [InlineData("double", "-0.0")]
+        [InlineData("decimal", "-0.0m")]
+        [InlineData("string", "\"\"")]
+        [InlineData("int?", "0")]
+        [InlineData("int?", "default(int)")]
+        [InlineData("DateTime?", "new DateTime()")]
+        [InlineData("DateTime?", "default(DateTime)")]
+        [InlineData("DateTimeOffset", "new DateTime()")]
+        [InlineData("DateTimeOffset", "default(DateTime)")]
+        public async Task TestNonDefaultValue(string type, string value)
+        {
+            string source = $@"
+using System;
+
+class C
+{{
+    void M(Span<{type}> span)
+    {{
+        span.Fill({value});
+    }}
+}}
+";
+            await TestCS(source);
+        }
+
+        [Fact]
+        public async Task TestGeneric_Unconstrained()
+        {
+            string source = @"
+using System;
+
+class C
+{
+    void M<T>(Span<T> span)
+    {
+        [|span.Fill(default)|];
+    }
+}
+";
+            await TestCS(source);
+        }
+
+        [Fact]
+        public async Task TestGeneric_Reference_Null()
+        {
+            string source = @"
+#nullable enable
+
+using System;
+
+class C
+{
+    void M<T>(Span<T?> span) where T : class
+    {
+        [|span.Fill(null)|];
+    }
+}
+";
+            await TestCS(source);
```

Comment:
Codefix should be tested here.

Reply:
The fixed result is too simple that I don't want to bother bloating the test code.

Reply:
> The fixed result is too simple that I don't want to bother bloating the test code.

I personally disagree. But deferring to @mavasani or @sharwell.